presto-iceberg/src/main/java/io/prestosql/plugin/iceberg/FilesTable.java

```diff
@@ -14,163 +14,107 @@
-                        .add(new ColumnMetadata("null_value_counts", typeManager.getType(mapType(INTEGER.getTypeSignature(), BIGINT.getTypeSignature()))))
-                        .add(new ColumnMetadata("lower_bounds", typeManager.getType(mapType(INTEGER.getTypeSignature(), VARCHAR.getTypeSignature()))))
-                        .add(new ColumnMetadata("upper_bounds", typeManager.getType(mapType(INTEGER.getTypeSignature(), VARCHAR.getTypeSignature()))))
-                        .add(new ColumnMetadata("key_metadata", VARBINARY))
-                        .add(new ColumnMetadata("split_offsets", new ArrayType(BIGINT)))
-                        .build());
-        this.snapshotId = requireNonNull(snapshotId, "snapshotId is null");
-    }
-
-    @Override
-    public Distribution getDistribution()
-    {
-        return Distribution.SINGLE_COORDINATOR;
-    }
-
-    @Override
-    public ConnectorTableMetadata getTableMetadata()
-    {
-        return tableMetadata;
-    }
-
-    @Override
-    public ConnectorPageSource pageSource(ConnectorTransactionHandle transactionHandle, ConnectorSession session, TupleDomain<Integer> constraint)
+    public static final IcebergColumnHandle FILE_PATH = new IcebergColumnHandle(-1, "file_path", VARCHAR, Optional.empty());
+    public static final IcebergColumnHandle FILE_FORMAT = new IcebergColumnHandle(-2, "file_format", VARCHAR, Optional.empty());
+    public static final IcebergColumnHandle RECORD_COUNT = new IcebergColumnHandle(-3, "record_count", BIGINT, Optional.empty());
+    public static final IcebergColumnHandle FILE_SIZE_IN_BYTES = new IcebergColumnHandle(-4, "file_size_in_bytes", BIGINT, Optional.empty());
+    public static final IcebergColumnHandle FILE_ORDINAL = new IcebergColumnHandle(-5, "file_ordinal", INTEGER, Optional.empty());
+    public static final IcebergColumnHandle SORT_COLUMNS = new IcebergColumnHandle(-6, "sort_columns", new ArrayType(INTEGER), Optional.empty());
+    public static final IcebergColumnHandle KEY_METADATA = new IcebergColumnHandle(-7, "key_metadata", VARBINARY, Optional.empty());
+    public static final IcebergColumnHandle SPLIT_OFFSETS = new IcebergColumnHandle(-8, "split_offsets", new ArrayType(BIGINT), Optional.empty());
+    public static final String COLUMN_SIZE = "column_size";
+    public static final String VALUE_COUNTS = "value_counts";
+    public static final String NULL_VALUE_COUNTS = "null_value_counts";
+    public static final String LOWER_BOUND = "lower_bound";
+    public static final String UPPER_BOUND = "upper_bound";
+    private final List<IcebergColumnHandle> columnHandles;
+
+    public FilesTable(Schema schema, TypeManager typeManager)
     {
-        return new FixedPageSource(buildPages(tableMetadata, session, icebergTable, snapshotId));
+        ImmutableList.Builder<IcebergColumnHandle> columnHandleBuilder = new ImmutableList.Builder<>();
```

Comment:
This can chain
```java
ImmutableList.Builder<IcebergColumnHandle> columnHandleBuilder = ImmutableList.builder()
        .add(FILE_PATH)
        .add(FILE_FORMAT)
        ...